### PR TITLE
Expanded instruments to include all GFZ indices

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Addresses #90 by adding the new GFZ Hpo and apo indices.  Added a general download function for GFZ data and expanded the existing F10.7 and SSN instruments to include the GFZ data as options.

# Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

```
import pysat
import pysatSpaceWeather as py_sw
import datetime as dt

hpo = pysat.Instrument(inst_module=py_sw.instruments.sw_hpo, tag='now', inst_id='30min')
stime = dt.datetime(2014, 3, 26)
hpo.download(start=stime)
hpo.load(date=stime)
hpo.data[:10]
```

Yields
```                      Hp30
2014-03-26 00:00:00  3.667
2014-03-26 00:30:00  4.667
2014-03-26 01:00:00  3.333
2014-03-26 01:30:00  3.333
2014-03-26 02:00:00  3.000
2014-03-26 02:30:00  2.667
2014-03-26 03:00:00  1.667
2014-03-26 03:30:00  1.333
2014-03-26 04:00:00  1.333
2014-03-26 04:30:00  1.667
```

~~Currently the apo downloads don't work because the query at GFZ is broken.  I contacted them and hopefully it will be fixed soon.~~ apo downloads have been fixed, GFZ responded very quickly.

## Test Configuration
* Operating system: OS X Big Sur
* Version number: Python 3.9
* Any details about your local setup that are relevant: develop branch of pysat

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the
release checklist on the pysat wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: pysatSpaceWeather/instruments/methods/lisird.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def build_lisird_url(lisird_data_name, start, stop):
 
 
 def download(date_array, data_path, local_file_prefix, local_date_fmt,
-             lisird_data_name, freq, update_files=True, fill_vals=None):
+             lisird_data_name, freq, update_files=False, fill_vals=None):
```
Comment (jklenzing) [inline pysatSpaceWeather/instruments/methods/lisird.py:95]: Why is this changed?

File: pysatSpaceWeather/instruments/sw_apo.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+Properties
+----------
+platform
+    'sw'
+name
+    'apo'
+tag
+    - 'now' Near real-time nowcast apo data from GFZ
+inst_id
+    '30min', '60min'
```
Comment (jklenzing) [inline pysatSpaceWeather/instruments/sw_apo.py:13]: There's a mixed style throughout this package in the top-level docstring.  Sometimes inst_id uses bullets, sometimes not.  tag generally does.  Using bullets would allow a bit more description, which we don't get when defining in the code.

File: pysatSpaceWeather/instruments/sw_f107.py [hunk 3]
```diff
@@ -15,9 +15,12 @@
 tag
     - 'historic' LASP F10.7 data (downloads by month, loads by day)
     - 'prelim' Preliminary SWPC daily solar indices
+    - 'now' A mist of nowcast and definitive values from GFZ
     - 'daily' Daily SWPC solar indices (contains last 30 days)
     - 'forecast' Grab forecast data from SWPC (next 3 days)
     - '45day' 45-Day Forecast data from the Air Force
+inst_id
+    '', 'obs', 'adj'
```
Comment (jklenzing) [inline pysatSpaceWeather/instruments/sw_f107.py:23]: A good spot for defining inst_ids! Adjacent? Adjusted? Adjunct?

**Review by jklenzing — COMMENTED** [review]
Coverage of new code looks good.  Some questions and suggestions on style.

File: pysatSpaceWeather/instruments/sw_f107.py [hunk 4]
```diff
@@ -15,9 +15,12 @@
 tag
     - 'historic' LASP F10.7 data (downloads by month, loads by day)
     - 'prelim' Preliminary SWPC daily solar indices
+    - 'now' A mist of nowcast and definitive values from GFZ
```
[the remark this answers is not in the document]
Comment (aburrell) [reply pysatSpaceWeather/instruments/sw_f107.py:18]:
```suggestion
    - 'now' A mix of nowcast and definitive values from GFZ
```

File: pysatSpaceWeather/instruments/sw_apo.py (hunk 2 above)
Comment (aburrell) [reply pysatSpaceWeather/instruments/sw_apo.py:13]: I like the bullets idea, will apply it.

File: pysatSpaceWeather/instruments/methods/lisird.py (hunk 1 above)
Comment (aburrell) [reply pysatSpaceWeather/instruments/methods/lisird.py:95]: Because that's the default everywhere else and also in the docstring here.

**Review by jklenzing — APPROVED** [review]
